### PR TITLE
[WIP] Load schema caches for all connections

### PR DESCRIPTION
**Pull request by gmcgibbon**
### Summary

Loads schema caches for each connection. This currently doesn't work for the same reasons as the multi-db fixtures PR (https://github.com/rails/rails/pull/33945).

### CODE REVIEW

**Comment by eileencodes** [comment]
> This currently doesn't work for the same reasons as the multi-db fixtures PR

Do you mean this doesn't work in master for the same reasons? Or that this new code doesn't work?

**Comment by gmcgibbon** [comment]
@eileencodes Sorry, I'll clarify:

TL;DR I mean this doesn't work on master for the same reasons. Once we find a fix for the fixtures PR problem, this PR should work. 

The schema cache initializer works if you apply [this fix](https://github.com/rails/rails/pull/33945#issuecomment-425990412) outlined in the fixtures PR. The fix seems to break other things like how we run migrations for each database in `rails db:migrate` as well as other potential side effects. It has to do with pooling and [this line](https://github.com/rails/rails/blob/f6a16106fa124defcdf748514264098178173e83/activerecord/lib/active_record/connection_handling.rb#L146) always being `"primary"`, I think.
